plugins/modules/purefa_info.py

```diff
@@ -166,9 +166,18 @@ def generate_default_dict(module, array):
                 default_info["encryption_algorithm"] = encryption.data_at_rest.algorithm
                 default_info["encryption_module_version"] = encryption.module_version
             eradication = array_data.eradication_config
-            default_info["eradication_days_timer"] = int(
-                eradication.eradication_delay / SEC_TO_DAY
-            )
+            if SUBS_API_VERSION in api_version:
+                default_info["eradication_safe_mode_disabled_days_timer"] = int(
+                    eradication.disabled_delay / SEC_TO_DAY
+                )
+                default_info["eradication_safe_mode_enabled_days_timer"] = int(
+                    eradication.enabled_delay / SEC_TO_DAY
+                )
+            eradication_delay = getattr(eradication, "eradication_delay", None)
+            if eradication_delay is not None:
+                default_info["eradication_days_timer"] = int(
+                    eradication_delay / SEC_TO_DAY
```

Comment:
this still needs `eradication.`

Reply:
this still needs addressing

Reply:
Are you referring to not using `getattr` and instead doing this?
```
if eradication.eradication_delay is not None:
  default_info["erdaication_days_timer"] = int (
    eradication.eradication_delay / SEC_TO_DAY
  )
```

Or using `getattr` but inline in the if statement like this?
```
if getattr(eradication, "eradication_delay", None) is not None:
  default_info["erdaication_days_timer"] = int (
    eradication.eradication_delay / SEC_TO_DAY
  )
```

My implementation was just "extracting" the `eradication_delay` attribute to a variable to use.

Reply:
My bad - misread the code. Looks good now